src/Interpreters/ActionsDAG.cpp

```diff
@@ -275,17 +276,12 @@ const ActionsDAG::Node & ActionsDAG::addFunctionImpl(
     /// If all arguments are constants, and function is suitable to be executed in 'prepare' stage - execute function.
     if (node.function_base->isSuitableForConstantFolding())
     {
-        ColumnPtr column;
+        size_t num_rows = 0;
 
-        if (all_const)
-        {
-            size_t num_rows = arguments.empty() ? 0 : arguments.front().column->size();
-            column = node.function->execute(arguments, node.result_type, num_rows, true);
-        }
-        else
-        {
-            column = node.function_base->getConstantResultForNonConstArguments(arguments, node.result_type);
-        }
+        for (auto & argument : arguments)
+            num_rows = std::min(num_rows, argument.column->size());
+
+        auto column = node.function->execute(arguments, node.result_type, num_rows, true);
```

Comment:
Doesn't it mean that we will run even heavy functions in the prepare stage? And how we will construct the arguments?